File: 2023-2024/sourcecode/mod2/lab9/RoomBasicsJava/app/src/main/java/com/example/roombasicsjava/repository/UserRepository.java
package com.example.roombasicsjava.repository;

import androidx.lifecycle.LiveData;

import com.example.roombasicsjava.data.User;
import com.example.roombasicsjava.data.UserDao;

import java.util.List;

public class UserRepository {
    private final UserDao userDao;

    public UserRepository(UserDao userDao) {
        this.userDao = userDao;
    }

    public LiveData<List<User>> getUsers() {
        return userDao.getUsers();
    }

    public void clear() {
        userDao.deleteAll();
    }

    public void add(User user) {
        userDao.insert(user);
    }
}

File: 2023-2024/sourcecode/mod2/lab9/RoomBasicsKotlin/app/src/main/java/com/example/roombasicskotlin/repository/UserRepository.kt
package com.example.roombasicskotlin.repository

import com.example.roombasicskotlin.data.User
import com.example.roombasicskotlin.data.UserDao

class UserRepository(private val userDao: UserDao) {
    fun getUsers() = userDao.getUsers()
    suspend fun clear() = userDao.deleteAll()
    suspend fun add(user: User) = userDao.insert(user)
}